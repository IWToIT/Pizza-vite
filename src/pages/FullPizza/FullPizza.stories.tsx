import type { Meta, StoryObj } from "@storybook/react";
import { FullPizza } from "./index";
import { Provider } from "react-redux";
import { store } from "@/redux/store";
import { BrowserRouter } from "react-router-dom";

const meta: Meta<typeof FullPizza> = {
  component: FullPizza,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Provider store={store}>
          <Story />
        </Provider>
      </BrowserRouter>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof FullPizza>;

export const Primary: Story = {};
