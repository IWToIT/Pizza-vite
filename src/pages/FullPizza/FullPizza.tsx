import React from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { useParams, useNavigate } from 'react-router-dom';
import { pizzaData } from '@/utils/consts';

const FullPizza: React.FC = () => {
  const [pizza, setPizza] = React.useState<{
    imageUrl: string;
    title: string;
    price: string;
  }>();

  const { id } = useParams();
  const navigate = useNavigate();

  React.useEffect(() => {
    async function fetchPizza() {
      try {
        const { data } = await axios.get(`${pizzaData}` + id);
        setPizza(data);
      } catch (error) {
        alert('Ошибка при получении пиццы!');
        navigate('/');
      }
    }

    fetchPizza();
  }, []);

  if (!pizza) {
    return <>Загрузка...</>;
  }

  return (
    <div className="container" data-testid="container-element">
      <img src={pizza.imageUrl} alt={pizza.title} />
      <h2>{pizza.title}</h2>
      <h4>{pizza.price} руб.</h4>
      <Link to="/">
        <button className="button button--outline button--add">
          <span>Назад</span>
        </button>
      </Link>
    </div>
  );
};

export default FullPizza;
