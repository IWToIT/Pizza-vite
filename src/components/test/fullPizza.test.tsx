import React from 'react';
import axios from 'axios';
import { render } from './test-utils';
import { screen, waitFor } from '@testing-library/react';
import { FullPizza } from '@/pages/FullPizza';

test('is pizza displayed when data is received successfully', async () => {
  const testData = {
    data: { imageUrl: 'testImageUrl', title: 'testTitle', price: 'testPrice' },
  };
  const title = testData.data.title;
  jest.spyOn(axios, 'get').mockResolvedValue(testData);
  render(<FullPizza />);

  await waitFor(() => expect(axios.get).toHaveBeenCalledTimes(1));

  expect(screen.getByTestId('container-element')).toBeInTheDocument();
  expect(screen.getByTestId('container-element')).toMatchSnapshot();
  expect(screen.getByAltText(title)).toBeInTheDocument();
  expect(screen.getByText(title)).toBeInTheDocument();
  expect(screen.getByText('testPrice руб.')).toBeInTheDocument();
});
