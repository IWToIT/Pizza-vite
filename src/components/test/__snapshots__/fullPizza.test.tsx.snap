// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is pizza displayed when data is received successfully 1`] = `
<div
  class="container"
  data-testid="container-element"
>
  <img
    alt="testTitle"
    src="testImageUrl"
  />
  <h2>
    testTitle
  </h2>
  <h4>
    testPrice
     руб.
  </h4>
</div>
`;

exports[`отображается ли пицца при успешном получении данных 1`] = `
<div
  class="container"
  data-testid="container-element"
>
  <img
    alt="testTitle"
    src="testImageUrl"
  />
  <h2>
    testTitle
  </h2>
  <h4>
    testPrice
     руб.
  </h4>
</div>
`;
